git-tar/function/ops.go

```diff
@@ -547,21 +550,22 @@ func importSecrets(pushEvent sdk.PushEvent, stack *stack.Services, clonePath str
 	return nil
 }
 
-func formatTemplateRepos() []string {
+func formatTemplateRepos() ([]string, error) {
 	templateRepos := []string{"https://github.com/openfaas/templates"}
 
+	var err error
 	if envTemplates := os.Getenv("custom_templates"); len(envTemplates) > 0 {
-		customTemplates := strings.Split(envTemplates, ",")
+		customTemplates := strings.Split(strings.Trim(envTemplates, " "), ",")
 		for _, repo := range customTemplates {
-			if strings.Contains(envTemplates, "https://") || strings.Contains(envTemplates, "http://") {
-				repo = strings.Trim(repo, " ")
-				templateRepos = append(templateRepos, repo)
+			repo = strings.Trim(repo, " ")
+			if _, err = url.ParseRequestURI(repo); err != nil {
```

Comment:
Looks like we could lose errors here because we don't return

Reply:
What happened with this?

Reply:
Yeh, I moved it to accumulate a list of errors, then check the list a little later on - im not 100% sure im happy with the solution in the long run, but I couldn't think of a better solution at the time.